### PR TITLE
fix(@nestjs/ng-universal) Adding try-catch block to `setupUniversal()`...

### PR DESCRIPTION
**Pull request by quinnjr**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

Adds a try-catch block to the `setupUniversal()` function to bubble-up underlying errors in `@nguniversal/express-engine`. Adds devDependencies that prevented local intergration testing and development. Adds fields to `package.json` for resolution of the module that was failing on MacOS.

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Behavior helps to fix issues on MacOS development that was preventing my team from developing and building the package locally.

Issue Number: https://github.com/nestjs/ng-universal/issues/838

## What is the new behavior?
`@nestjs/ng-universal` now bubbles-up error messages and is locally testable for integration.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

… Adding `@angular` devDependencies for local development. Adding `HttpAdapterHost` to `AngularUniversalModule.forRoot`.

This fix helps to resolve https://github.com/nestjs/ng-universal/issues/838

1. Adding `@angular` devDependencies for local development.

Without the addition of a few `@angular` and associated projects
dependencies to devDependencies, local filesystem development of
`@nestjs/ng-universal` would throw error messages during compilation:

```sh
../ng-universal/dist/angular-universal.module.js:31:0-27 - Error: Module not found: Error: Can't resolve 'reflect-metadata' in '/home/joseph/ng-universal/dist'

../ng-universal/node_modules/@nestjs/common/index.js:11:0-27 - Error: Module not found: Error: Can't resolve 'reflect-metadata' in '/home/joseph/ng-universal/node_modules/@nestjs/common'

../ng-universal/node_modules/@nestjs/common/pipes/validation.pipe.js:55:164-190 - Error: Module not found: Error: Can't resolve 'class-validator' in '/home/joseph/ng-universal/node_modules/@nestjs/common/pipes'

../ng-universal/node_modules/@nestjs/common/pipes/validation.pipe.js:59:172-200 - Error: Module not found: Error: Can't resolve 'class-transformer' in '/home/joseph/ng-universal/node_modules/@nestjs/common/pipes'

../ng-universal/node_modules/@nestjs/common/serializer/class-serializer.interceptor.js:22:239-267 - Error: Module not found: Error: Can't resolve 'class-transformer' in '/home/joseph/ng-universal/node_modules/@nestjs/common/serializer'

../ng-universal/node_modules/@nestjs/core/index.js:11:0-27 - Error: Module not found: Error: Can't resolve 'reflect-metadata' in '/home/joseph/ng-universal/node_modules/@nestjs/core'

../ng-universal/node_modules/@angular/common/fesm2015/common.mjs:6:0-36 - Error: Module not found: Error: Can't resolve '@angular/core' in '/home/joseph/ng-universal/node_modules/@angular/common/fesm2015'

../ng-universal/node_modules/@angular/common/fesm2015/http.mjs:8:0-36 - Error: Module not found: Error: Can't resolve '@angular/core' in '/home/joseph/ng-universal/node_modules/@angular/common/fesm2015'

../ng-universal/node_modules/@angular/platform-server/fesm2015/platform-server.mjs:7:0-36 - Error: Module not found: Error: Can't resolve '@angular/core' in '/home/joseph/ng-universal/node_modules/@angular/platform-server/fesm2015'

../ng-universal/node_modules/@angular/platform-server/fesm2015/platform-server.mjs:9:0-48 - Error: Module not found: Error: Can't resolve '@angular/platform-browser' in '/home/joseph/ng-universal/node_modules/@angular/platform-server/fesm2015'

../ng-universal/node_modules/@angular/platform-server/fesm2015/platform-server.mjs:12:0-63 - Error: Module not found: Error: Can't resolve '@angular/animations/browser' in '/home/joseph/ng-universal/node_modules/@angular/platform-server/fesm2015'

../ng-universal/node_modules/@angular/platform-server/fesm2015/platform-server.mjs:14:0-73 - Error: Module not found: Error: Can't resolve '@angular/platform-browser-dynamic' in '/home/joseph/ng-universal/node_modules/@angular/platform-server/fesm2015'

../ng-universal/node_modules/@angular/platform-server/fesm2015/platform-server.mjs:15:0-103 - Error: Module not found: Error: Can't resolve '@angular/platform-browser/animations' in '/home/joseph/ng-universal/node_modules/@angular/platform-server/fesm2015'

../ng-universal/node_modules/@angular/platform-server/fesm2015/platform-server.mjs:19:0-61 - Error: Module not found: Error: Can't resolve '@angular/compiler' in '/home/joseph/ng-universal/node_modules/@angular/platform-server/fesm2015'

../ng-universal/node_modules/@nguniversal/express-engine/fesm2015/tokens.mjs:1:0-47 - Error: Module not found: Error: Can't resolve '@angular/core' in '/home/joseph/ng-universal/node_modules/@nguniversal/express-engine/fesm2015'
```

2. Adding `HttpAdapterHost` to `AngularUniversalModule.forRoot`

Without adding `HttpAdapterHost` to the providers array return with the
dynamic module from `AngularUniversalModule.forRoot`,
`@nestjs/ng-universal` would complain about the missing provider during
compilation:

```sh
[Nest] 4992  - 04/08/2022, 8:01:08 AM   ERROR [ExceptionHandler] Nest can't resolve dependencies of the AngularUniversalModule (ANGULAR_UNIVERSAL_OPTIONS, ?). Please make sure that the argument HttpAdapterHost at index [1] is available in the AngularUniversalModule context.

Potential solutions:
- If HttpAdapterHost is a provider, is it part of the current AngularUniversalModule?
- If HttpAdapterHost is exported from a separate @Module, is that module imported within AngularUniversalModule?
  @Module({
    imports: [ /* the Module containing HttpAdapterHost */ ]
  })

Error: Nest can't resolve dependencies of the AngularUniversalModule (ANGULAR_UNIVERSAL_OPTIONS, ?). Please make sure that the argument HttpAdapterHost at index [1] is available in the AngularUniversalModule context.

Potential solutions:
- If HttpAdapterHost is a provider, is it part of the current AngularUniversalModule?
- If HttpAdapterHost is exported from a separate @Module, is that module imported within AngularUniversalModule?
  @Module({
    imports: [ /* the Module containing HttpAdapterHost */ ]
  })

    at /home/joseph/nestjs-macos-repro/dist/nestjs-macos-repro/server/main.js:10293:15
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/home/joseph/nestjs-macos-repro/dist/nestjs-macos-repro/server/main.js:270808:24)
    at _next (/home/joseph/nestjs-macos-repro/dist/nestjs-macos-repro/server/main.js:270830:9)
    at _ZoneDelegate.invoke (/home/joseph/nestjs-macos-repro/dist/nestjs-macos-repro/server/main.js:116388:26)
    at Zone.run (/home/joseph/nestjs-macos-repro/dist/nestjs-macos-repro/server/main.js:116150:43)
    at /home/joseph/nestjs-macos-repro/dist/nestjs-macos-repro/server/main.js:117222:36
    at _ZoneDelegate.invokeTask (/home/joseph/nestjs-macos-repro/dist/nestjs-macos-repro/server/main.js:116422:31)
    at Zone.runTask (/home/joseph/nestjs-macos-repro/dist/nestjs-macos-repro/server/main.js:116194:47)
    at drainMicroTaskQueue (/home/joseph/nestjs-macos-repro/dist/nestjs-macos-repro/server/main.js:116601:35)

A server error has occurred.
node exited with 1 code.
connect ECONNREFUSED 127.0.0.1:43387
```

3. Adding try-catch block to `setupUniversal()`

Without the try-catch block (now implemented) in `setupUniversal()`,
NestJS would silently fail and exit with improper configurations between
NestJS and the underlying `@nguniversal/express-engine` package. The new
try-catch block provides the missing information for error checking and
debugging.

### CODE REVIEW

**Comment by quinnjr** [comment]
Related support thread: https://discord.com/channels/520622812742811698/520649487924985885/960529416092524554

**Comment by quinnjr** [comment]
@DaSchTour can you add this to your fork?

**Comment by DaSchTour** [comment]
@quinnjr I added it to `@nestjs-fixed/ng-universal@6.2.4`

File: lib/angular-universal.module.ts [hunk 1]
```diff
@@ -36,7 +36,8 @@ export class AngularUniversalModule implements OnModuleInit {
         {
           provide: ANGULAR_UNIVERSAL_OPTIONS,
           useValue: options
-        }
+        },
+        HttpAdapterHost
```
Comment (kamilmysliwiec) [inline lib/angular-universal.module.ts:40]: HttoAdapterHost should never be re-registered by non-internal packages

**Pull request closed without merging**